src/org/ods/util/HelmStatusSimpleData.groovy

```diff
@@ -0,0 +1,186 @@
+package org.ods.util
+
+import com.cloudbees.groovy.cps.NonCPS
+import groovy.transform.TypeChecked
+
+@TypeChecked
+class HelmStatusSimpleData {
+
+    String name
+    String version
+    String namespace
+    String status
+    String description
+    String lastDeployed
+    /**
+     * Resources names by kind
+     */
+    Map<String, List<String> > resourcesByKind
+
+    @SuppressWarnings(['Instanceof'])
+    @NonCPS
+        static HelmStatusSimpleData fromJsonObject(Object object) {
+        try {
+            def rootObject = ensureMap(object, "")
+
+            def infoObject = ensureMap(rootObject.info, "info")
+
+            // validation of missing keys or types
+            def missingKeys = []
+            def badTypes = []
+            def expectedStringAttribsForRoot = collectMissingStringAttributes(rootObject,
```

Comment:
It is not worth it to validate the whole Map. We need to validate that the values we use satisfy the constraints of this class. Otherwise, if there is a problem (such as something that should be a List isn't) we can trust runtime exception to happen automatically.

Reply:
This is used for creating a message reporting the missing keys and bad types. 
This message is used to instantiate a new IllegalArgumentException(message), in order to report those errors on an exception trace.